c/driver/framework/CMakeLists.txt

```diff
@@ -35,6 +36,7 @@ if(ADBC_BUILD_TESTS)
                 base_driver_test.cc
                 EXTRA_LINK_LIBS
                 adbc_driver_framework)
+  target_compile_definitions(adbc-driver-framework-test PRIVATE ADBC_STATIC)
```

Comment:
I don't see the comment I thought I left originally but I wanted to check this was the desired behavior. The way the macro is now, the library itself and anything that links to it must define ADBC_STATIC. The former makes sense, but we don't want the downstream target to have to define that, right? @kou

Reply:
Hmm. Not right. All downstream targets also need to define `ADBC_STATIC`. We can automate it by changing  `target_compile_definitions(adbc_driver_framework PRIVATE ADBC_EXPORTING ADBC_STATIC)` to:

```cmake
target_compile_definitions(adbc_driver_framework PRIVATE ADBC_EXPORTING)
target_compile_definitions(adbc_driver_framework PUBLIC ADBC_STATIC)
```

Reply:
Ah thanks - makes sense

Reply:
Ah, we should use separated `ADBC_XXX_STATIC` like `ADBC_XXX_EXPORT`.